ads/google/a4a/utils.js

```diff
@@ -960,3 +962,50 @@ export function getAmpRuntimeTypeParameter(win) {
   const art = getBinaryTypeNumericalCode(getBinaryType(win));
   return isCdnProxy(win) && art != '0' ? art : null;
 }
+
+/**
+ * Returns the fixed size of the given element, or the fixed size of its nearest
+ * ancestor that has a fixed size, if the given element has none.
+ * @param {!Window} win
+ * @param {?Element} element
+ * @return {number} The width of the given element, or of the nearest ancestor
+ *    with a fixed size, if the given element has none.
+ */
+export function getContainerWidth(win, element) {
+  let el = element;
+  let maxDepth = 100;
+  // Find the first ancestor with a fixed size
+  while (el && maxDepth--) {
+    const layout = el.getAttribute('layout');
+    switch (layout) {
+      case Layout.FIXED:
+        return parseInt(el.getAttribute('width'), 10) || 0;
+      case Layout.RESPONSIVE:
+      case Layout.FILL:
+      case Layout.FIXED_HEIGHT:
+      case Layout.FLUID:
+        // The above layouts determine the width of the element by the
+        // containing element, or by CSS max-width property.
+        const maxWidth = parseInt(computedStyle(win, el).maxWidth, 10);
```

Comment:
Is calling computedStyle ok here?

Reply:
There's no strict rule against it, and I don't think this loop will iterate more than twice in practice.

Reply:
Talked to @glevitzky offline. We already have code calculating `computedStyle` from every nested parent [here](https://github.com/ampproject/amphtml/blob/master/src/ad-helper.js#L56). It would be great if we can leverage the logic there later. But I'm fine with us calling `computedStyle` here if the `getContainerWidth` is only called once.

Reply:
Technically, we call the method twice, but the second call checks at most one element (different from any of the elements checked in the first call).